### PR TITLE
Add handling for non-public visibility

### LINKED ISSUE #13: Handle visibilities in a more elegant way
Currently, `sealed-enum` expects and generates `public` implementations of `SealedEnum` and extension methods.

This leads to compilation failures if the sealed class (or any of its sealed class subclasses) are `private`, `protected` or `internal`.
Additionally, this will lead to compilation failures if the objects are `private` or `protected`.

For `private` or `protected` (in either case), we should surface an error during annotation processing as we won't have the needed access.

For `internal`, we should ensure that any generated types are also internal to allow compilation. The existing behavior for `internal` objects already permits compliation, since we do not expose their types anywhere in the generated code.

### PR DESCRIPTION
**Pull request by alexvanyo**
The current functionality of `sealed-enum` assumes that the sealed class and companion object were both public. This PR adds handling for internal sealed classes and companion objects, as well as explicit errors for private or protected classes, and thus fixes #13.

This PR has two commits:

- The first just reorganizes all of the `*Spec` files into a `spec` package

- The second adds the visibility handling support

### CODE REVIEW

File: sealedenumprocessor/src/main/kotlin/com/livefront/sealedenum/internal/Visibility.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.livefront.sealedenum.internal
+
+import com.squareup.kotlinpoet.KModifier
+
+/**
+ * A restricted enumeration of visibilities for use with sealed classes. Due to generating code that is in a different
+ * file, we can only reasonably work with sealed classes and companion objects that are public or internal.
```
Comment (byencho) [inline sealedenumprocessor/src/main/kotlin/com/livefront/sealedenum/internal/Visibility.kt:7]: 👍

File: sealedenumprocessor/src/main/kotlin/com/livefront/sealedenum/internal/spec/EnumForSealedEnumTypeSpec.kt [hunk 2]
```diff
@@ -10,21 +11,23 @@ import javax.lang.model.element.TypeElement
...
     private val sealedClass: SealedClass,
+    private val sealedClassVisibility: Visibility,
     private val sealedClassCompanionObjectElement: TypeElement,
     private val sealedObjects: List<SealedObject>,
     private val parameterizedSealedClass: TypeName,
     private val enumPrefix: String,
     private val sealedClassInterfaces: List<TypeName>
 ) {
     private val typeSpecBuilder = TypeSpec.enumBuilder(sealedClass.createEnumForSealedEnumName(enumPrefix))
+        .addModifiers(sealedClassVisibility.kModifier)
```
Comment (byencho) [inline sealedenumprocessor/src/main/kotlin/com/livefront/sealedenum/internal/spec/EnumForSealedEnumTypeSpec.kt:30]: So previously this was just always an implied `public` right?

**Review by byencho — APPROVED** [review]
Once again, very throughly tested! Very nice. Looks good to me 👍

File: sealedenumprocessor/src/main/kotlin/com/livefront/sealedenum/internal/spec/EnumForSealedEnumTypeSpec.kt (hunk 2 above)
Comment (alexvanyo) [reply sealedenumprocessor/src/main/kotlin/com/livefront/sealedenum/internal/spec/EnumForSealedEnumTypeSpec.kt:30]: Correct, adding no modifiers resulted in an implied `public`. kotlinpoet is also smart enough to not emit the `public` in the generated code even when we do specify it.

**Comment by alexvanyo** [comment]
Thanks @byencho !

**Pull request merged**